e2e/playwright/actions/src/tests/share/share-file.spec.ts

```diff
@@ -0,0 +1,246 @@
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ApiClientFactory, NodesApi, test, timeouts, Utils, SharedLinksApi } from '@alfresco/playwright-shared';
+import { expect } from '@playwright/test';
+
+test.describe('Share a file', () => {
+  const random = Utils.random();
+
+  const username = `user-${random}`;
+  const parent = `parent-${random}`;
+  let parentId: string;
+
+  const file3 = `file3-${random}.txt`;
+  const file4 = `file4-${random}.txt`;
+  const file5 = `file5-${random}.txt`;
+  const file6 = `file6-${random}.txt`;
+  const file7 = `file7-${random}.txt`;
+  const file8 = `file8-${random}.txt`;
+  const file9 = `file9-${random}.txt`;
+
+  const shareLinkPreUrl = `/#/preview/s/`;
+
+  const apiClientFactory = new ApiClientFactory();
+
+  test.beforeAll(async () => {
+    await apiClientFactory.setUpAcaBackend('admin');
+    await apiClientFactory.createUser({ username });
+    const nodesApi = await NodesApi.initialize(username, username);
+
+    parentId = (await nodesApi.createFolder(parent)).entry.id;
+  });
+
+  test.afterAll(async () => {
+    await apiClientFactory.nodes.deleteNodes([parentId]);
+  });
+
+  test.describe('when logged out', () => {
+    let file6SharedLink: string;
+    let file6Id: string;
+
+    test.beforeAll(async () => {
+      const nodesApi = await NodesApi.initialize(username, username);
+      const shareApi = await SharedLinksApi.initialize(username, username);
+
+      file6Id = (await nodesApi.createFile(file6, parentId))?.entry.id;
+
+      const sharedId = (await shareApi.shareFileById(file6Id)).entry.id;
+      file6SharedLink = `${shareLinkPreUrl}${sharedId}`;
+      await shareApi.waitForFilesToBeShared([file6Id]);
+    });
+
+    test.afterAll(async () => {
+      await apiClientFactory.nodes.deleteNodes([file6Id]);
+    });
+
+    test('[C286326] A non-logged user can download the shared file from the viewer', async ({ personalFiles, page }) => {
+      await page.goto(file6SharedLink);
+      await personalFiles.viewer.waitForViewerToOpen();
+
+      const downloadPromise = personalFiles.page.waitForEvent('download');
+      await personalFiles.viewer.toolbar.sharedDownloadButton.click();
+      const download = await downloadPromise;
+      expect(download.suggestedFilename()).toBe(file6);
+    });
+  });
+
+  test.describe('when logged in', () => {
+    const expiryDateObj: Date = new Date();
```

Comment:
You don't have to specify a type when you're creating the object right away

Reply:
removed type declaration